lib/api/s3/container.ex

```diff
@@ -3,20 +3,46 @@ defmodule FileStorageApi.API.S3.Container do
   @behaviour FileStorageApi.Container
 
   import FileStorageApi.API.S3.Base
+
   alias ExAws.S3
-  alias FileStorageApi.{Container, File}
+  alias FileStorageApi.Container
+  alias FileStorageApi.File
 
   @impl true
-  def create(container_name, options \\ %{}) do
+  def create(container_name, connection_name, options \\ %{}) do
     result =
       container_name
       |> S3.put_bucket(region())
-      |> request()
+      |> request(connection_name)
 
     case result do
       {:ok, result} ->
-        put_public_policy(container_name, options)
```

Comment:
Why not just add an extra variable that making a really big function now.

Reply:
I don't like having this connection-name variable dribble into all kinds of functions, while those functions don't use the value but just pass it along to the next function.

Since `request` is the one that needs to have the variable, I've moved all invocations of `request` into the `create`-function.

I like the result: `create` is the main function that does all the checks, and depending on the result it calls the next function.
I do agree that it is now too big, so I've refactored it a little to make it smaller.